### ISSUE TITLE
Use only one way of handling search string

### ISSUE THREAD

**Issue by kirillku** [Kelin2025/atomic-router#8]
Currently, there are two ways of handling search string:
- `URLSearchParams` in [`matchPath`](https://github.com/Kelin2025/atomic-router/blob/main/src/build-path.ts#L17)
- `qs` (`parse`) in [`createHistoryRouter`](https://github.com/Kelin2025/atomic-router/blob/main/src/create-history-router.ts#L160)

Only one should be used to avoid possible errors. They work slightly different. Handling of basic strings is the same, but the rest is different.

You can change this way, then `qs` dependency will be not needed anymore. (There will be only basic stuff handling, but complex stuff is not handled right now anyway)
```diff
- parse(history.location.search.slice(1)) as RouteQuery,
+ Object.fromEntries(new URLSearchParams(history.location.search))
```

**Comment by Kelin2025** [comment]
Honestly I think we should get rid of `route.$query` at all and move it to `router`
There are query params that might be common for all routes - for example, `/any-page?utm_source=medium`, and for now we should access them through `someRoute.$query`
However, there also might be route-specific query params (like `/posts?filter_by=user`). And I wonder how we should treat these ones, 'cause for now they are not typed

**Comment by Kelin2025** [comment]
Fixed in `v0.4.0` with a couple of new features
https://github.com/Kelin2025/atomic-router/releases/tag/v0.4.0
